libs/wingsdk/src/target-sim/queue.ts

```diff
@@ -71,13 +72,13 @@ export class Queue extends cloud.Queue implements ISimulatorResource {
       props
     );
 
-    // At the time the queue is created in the simulator, it needs to be able to
-    // call subscribed functions.
-    this.node.addDependency(fn);
-
-    this.subscribers.push({
-      functionHandle: simulatorHandleToken(fn),
-      batchSize: props.batchSize ?? 1,
+    new EventMapping(this, `${this.node.id}-QueueEventMapping-${hash}`, {
+      subscriber: fn,
+      publisher: this,
+      eventSubscription: {
+        functionHandle: simulatorHandleToken(fn),
```

Comment:
Can we remove this field `functionHandle` since we already pass `fn` as a subscriber?

Reply:
I dont know if we can without making the assumption that every subscription will have the `functionHandle` property. Which for now is always true but what Im thinking of is like event mapping between a topic and a queue (which does not require an intermediate function)

Reply:
I wouldn't overdesign for other kinds of event mappings yet. If it's a concern, something you could consider is adding a field to the `eventSubscription` to indicate what the type of the mapping is or what the types of the resources are.

Since every event mapping knows who the subscriber is, the inflight code should be able to pass it along to all subscribers I think? eg
```ts
const client = this.context.findInstance(this.publisher) as IEventPublisher;
await client.addEventSubscription(this.subscriber, this.eventSubscription);
```

Reply:
I think Im on the same page here, made some updates let me know if they align.